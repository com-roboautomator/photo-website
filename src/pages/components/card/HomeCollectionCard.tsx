import {createStyles, Link, withStyles, WithStyles} from '@material-ui/core'
import React from 'react'
import '../../../assets/css/OpenSans-Light.ttf'
import collection, {placeholder} from '../../../assets/data/ImageDataStructure'
import HomeCollectionCardImage from './HomeCollectionCardImage'
import HomeCollectionCardText from './HomeCollectionCardText'

interface CollectionCardProps extends WithStyles<typeof styles> {
    key?: string
    collection: collection
    height: number
    selected: boolean
}

class CollectionCard extends React.Component<CollectionCardProps> {
    render() {
        const classes = this.props.classes
        const collection = this.props.collection
        const hrefLink: string = `gallery/${this.props.collection.key}`
        const imageURL: string =
            this.props.collection.images === undefined
                ? placeholder.url
                : this.props.collection.images[
                      this.props.collection.titleImage ?? 0
                  ].url
        return (
            <div
                data-testid={`Home-Collection-Card-${this.props.collection.title}`}
                className={
                    this.props.selected
                        ? classes.spacing_selected
                        : classes.spacing_unselected
                }>
                <div
                    className={classes.container}
                    data-testid="Home-Collection-Card-Container">
                    <Link href={hrefLink} style={{textDecoration: 'none'}}>
                        <HomeCollectionCardImage
                            src={imageURL}
                            height={this.props.height - 70} // remove the text height from image value
                        />
                        <HomeCollectionCardText
                            title={collection.title}
                            tagTitle={collection.tagTitle}
                            tagColour={collection.tagColour}
                        />
                    </Link>
                </div>
            </div>
        )
    }
}

const styles = () =>
    createStyles({
        spacing_selected: {
            pointerEvents: 'auto',
            userSelect: 'none',
            position: 'relative',
            zIndex: 99,
            WebkitTransition: 'opacity 0.7s, transform 0.4s, margin 0.4s',
            transform: 'scale(1)',
            opacity: '100%',
            flex: 1,
        },
        spacing_unselected: {
            pointerEvents: 'auto',
            userSelect: 'none',
            flex: 1,
            WebkitTransition: 'opacity 0.7s, transform 0.4s, margin 0.4s',
            transform: 'scale(0.8)',
            opacity: '90%',
        },
        container: {
            width: 350,
            borderTopLeftRadius: '25px',
            borderTopRightRadius: '25px',
            borderBottomLeftRadius: '25px',
            borderBottomRightRadius: '25px',
            boxShadow: '1px 2px 10px rgba(0, 0, 0, 0.6)',
        },
    })

export default withStyles(styles)(CollectionCard)
