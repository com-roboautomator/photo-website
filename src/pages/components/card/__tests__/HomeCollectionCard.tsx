import {render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import HomeCollectionCard from '../HomeCollectionCard'

describe('Home Collection Card', () => {
    it('should render collection card spacing correctly when unselected', () => {
        const component = render(
            <HomeCollectionCard
                collection={{
                    key: 'key',
                    title: 'Home-Collection-Card-Title',
                    index: 0,
                }}
                selected={false}
            />
        )

        expect(component).toBeTruthy()
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveClass(
            'CollectionCard-spacing_unselected-2'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'pointerEvents: auto'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'userSelect: none'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'flex: 1'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'transform: scale(0.8)'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'opacity: 90%'
        )
    })

    it('should render collection card spacing correctly when selected', () => {
        const component = render(
            <HomeCollectionCard
                collection={{
                    key: 'key',
                    title: 'Home-Collection-Card-Title',
                    index: 0,
                }}
                selected={true}
            />
        )

        expect(component).toBeTruthy()
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveClass(
            'CollectionCard-spacing_selected-11'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'pointerEvents: auto'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'userSelect: none'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'flex: 1'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'zIndex: 99'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'transform: scale(1)'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'opacity: 100%'
        )
        expect(component.getByTestId('Collection-Card-Spacing')).toHaveStyle(
            'position: relative'
        )
    })

    it('should render card image correctly', () => {
        const component = render(
            <HomeCollectionCard
                collection={{
                    key: 'key',
                    title: 'Home-Collection-Card-Title',
                    index: 0,
                }}
                selected={true}
            />
        )

        expect(component).toBeTruthy()
        expect(component.getByTestId('Collection-Card-Image')).toHaveAttribute(
            'src',
            'https://picsum.photos/id/250/500'
        )
        expect(component.getByTestId('Collection-Card-Image')).toHaveAttribute(
            'width',
            '350'
        )
        expect(component.getByTestId('Collection-Card-Image')).toHaveAttribute(
            'height',
            '200'
        )
        expect(component.getByTestId('Collection-Card-Image')).toHaveClass(
            'CollectionCardImage-layout-24'
        )
        expect(component.getByTestId('Collection-Card-Image')).toHaveStyle(
            'borderTopLeftRadius: 25px'
        )
        expect(component.getByTestId('Collection-Card-Image')).toHaveStyle(
            'borderTopRightRadius: 25px'
        )
    })

    it('should render card text wrapper correctly', () => {
        const component = render(
            <HomeCollectionCard
                collection={{
                    key: 'key',
                    title: 'Home-Collection-Card-Title',
                    index: 0,
                }}
                selected={true}
            />
        )

        expect(component).toBeTruthy()
        expect(
            component.getByTestId('Collection-Card-Text-Wrapper')
        ).toHaveClass('CollectionCardText-text_wrapper-36')
        expect(
            component.getByTestId('Collection-Card-Text-Wrapper')
        ).toHaveStyle('position: relative')
        expect(
            component.getByTestId('Collection-Card-Text-Wrapper')
        ).toHaveStyle('zIndex: 99')
        expect(
            component.getByTestId('Collection-Card-Text-Wrapper')
        ).toHaveStyle('flex: 1')
        expect(
            component.getByTestId('Collection-Card-Text-Wrapper')
        ).toHaveStyle('flexDirection: row')
        expect(
            component.getByTestId('Collection-Card-Text-Wrapper')
        ).toHaveStyle('fontSize: 100%')
        expect(
            component.getByTestId('Collection-Card-Text-Wrapper')
        ).toHaveStyle('color: black')
        expect(
            component.getByTestId('Collection-Card-Text-Wrapper')
        ).toHaveStyle('borderBottomLeftRadius: 25px')
        expect(
            component.getByTestId('Collection-Card-Text-Wrapper')
        ).toHaveStyle('borderBottomRightRadius: 25px')
        expect(
            component.getByTestId('Collection-Card-Text-Wrapper')
        ).toHaveStyle('borderTop: 1px solid gray')
        expect(
            component.getByTestId('Collection-Card-Text-Wrapper')
        ).toHaveStyle('height: 45px')
    })

    it('should render card image icon correctly', () => {
        const component = render(
            <HomeCollectionCard
                collection={{
                    key: 'key',
                    title: 'Home-Collection-Card-Title',
                    index: 0,
                }}
                selected={true}
            />
        )

        expect(component).toBeTruthy()
        expect(
            component.getByTestId('Collection-Card-Image-Icon')
        ).toHaveAttribute('src', 'collection-image-o.png')
        expect(
            component.getByTestId('Collection-Card-Image-Icon')
        ).toHaveAttribute('width', '20')
        expect(
            component.getByTestId('Collection-Card-Image-Icon')
        ).toHaveAttribute('height', '20')
    })

    it('should render card title correctly', () => {
        const component = render(
            <HomeCollectionCard
                collection={{
                    key: 'key',
                    title: 'test-collection-card-title',
                    index: 0,
                }}
                selected={true}
            />
        )

        expect(component).toBeTruthy()
        expect(
            component.getByTestId('Collection-Card-Title')
        ).toHaveTextContent('test-collection-card-title')
        expect(component.getByTestId('Collection-Card-Title')).toHaveClass(
            'CollectionCardText-text-58'
        )
        expect(component.getByTestId('Collection-Card-Title')).toHaveStyle(
            'fontFamily: Open-Sans-Regular'
        )
        expect(component.getByTestId('Collection-Card-Title')).toHaveStyle(
            'fontStyle: normal'
        )
        expect(component.getByTestId('Collection-Card-Title')).toHaveStyle(
            'textAlign: left'
        )
        expect(component.getByTestId('Collection-Card-Title')).toHaveStyle(
            'position: relative'
        )
        expect(component.getByTestId('Collection-Card-Title')).toHaveStyle(
            'paddingLeft: 10px'
        )
        expect(component.getByTestId('Collection-Card-Title')).toHaveStyle(
            'width: 75%'
        )
        expect(component.getByTestId('Collection-Card-Title')).toHaveStyle(
            'alignSelf: center'
        )
    })
})
