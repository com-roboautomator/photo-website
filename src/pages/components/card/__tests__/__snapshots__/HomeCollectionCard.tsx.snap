// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Collection Card should match snapshot 1`] = `
<div
  className="CollectionCard-spacing_unselected-2"
  data-testid="Home-Collection-Card-test-collection-title"
>
  <div
    className="CollectionCard-container-3"
    data-testid="Home-Collection-Card-Container"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="gallery/test-collection-key"
      onBlur={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <img
        alt=""
        className="CollectionCardImage-layout-4"
        data-testid="Collection-Card-Image"
        height={130}
        src="https://picsum.photos/id/250/500"
        width="350"
      />
      <div
        className="CollectionCardText-text_wrapper-6"
        data-testid="Collection-Card-Text-Wrapper"
      >
        <img
          alt=""
          className="CollectionCardText-icon-7"
          data-testid="Collection-Card-Image-Icon"
          height={20}
          src="collection-image-o.png"
          width={20}
        />
        <p
          className="CollectionCardText-text-8"
          data-testid="Collection-Card-Title"
        >
          test-collection-title
        </p>
        <div
          className="CollectionCardText-tag-9"
        >
          <div
            className="HomeCollectionTag-layout-10"
            data-testid="tag-layout"
            style={
              Object {
                "background": "#323232",
              }
            }
          >
            Default
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
